Libplanet/Blocks/RawBlock.cs

```diff
@@ -47,21 +50,27 @@ namespace Libplanet.Blocks
             Hash = hash;
         }
 
-        internal RawBlock(SerializationInfo info, StreamingContext context)
-            : this(
-                index: info.GetInt64("index"),
-                timestamp: info.GetString("timestamp"),
-                nonce: info.GetValue<byte[]>("nonce"),
-                miner: info.GetValue<byte[]>("reward_beneficiary"),
-                difficulty: info.GetInt32("difficulty"),
-                previousHash: info.GetValueOrDefault<byte[]>(
-                    "previous_hash",
-                    null
-                ),
-                transactions: info.GetValue<IEnumerable>("transactions"),
-                hash: info.GetValue<byte[]>("hash")
-            )
+        public RawBlock(Bencodex.Types.Dictionary dict)
         {
+            Func<string, byte[]> b = Encoding.ASCII.GetBytes;
+            Index = dict.GetValue<Integer>(b("index"));
+            Timestamp = dict.GetValue<Text>(b("timestamp"));
+            Difficulty = dict.GetValue<Integer>(b("difficulty"));
+            Transactions = dict.GetValue<Bencodex.Types.List>(b("transactions"))
+                .Select(tx => (byte[])(Binary)tx);
+            Nonce = dict.GetValue<Binary>(b("nonce"));
+
+            Miner = dict.ContainsKey((Binary)b("reward_beneficiary"))
```

Comment:
We might be better to define constants for these keys, e.g.:

```csharp
public static Binary RewardBeneficiaryKey = (Binary)b("reward_beneficiary");
```

@moreal Also we'd better to define an alternative constructor for `Bencodex.Types.Binary` to take a `string` and an `Encoding`, so that we can make a `Binary` value by calling `new Binary("reward_beneficiary", Encoding.ASCII)`?

Reply:
@limebell @moreal Sent a pull request to add a new constructor: <https://github.com/planetarium/bencodex.net/pull/14>.